swag/swag_client.py

```diff
@@ -369,3 +416,294 @@ async def execute_swagdmin_command(self, message):
                 "du serveur\n"
                 "```"
             )
+
+    async def execute_cagnotte_command(self, message):
+        def get_cagnotte_idx_from_command(splited_command):
+            cagnotte_idx = int(
+                [
+                    identifiant[1:]
+                    for identifiant in splited_command
+                    if identifiant.startswith("€") and identifiant[1:].isnumeric()
+                ][0]
+            )
+            return cagnotte_idx
+
+        message_command = message.content
+        splited_command = message_command.split()
+
+        if "créer $wag" in message_command:
+            cagnotte_name = " ".join(splited_command[3:])
+            if len(cagnotte_name) == 0:
+                await message.channel.send(
+                    "Merci de mentionner un nom pour ta €agnotte."
+                )
+                return
+            self.swag_bank.create_cagnotte(
+                cagnotte_name, Currency.SWAG, message.author.id
+            )
+
+            cagnotte_info = self.swag_bank.swagdb.get_cagnotte(cagnotte_name).get_info()
+            await message.channel.send(
+                f"{message.author.mention} vient de créer une €agnotte de $tyle nommée **« {cagnotte_name} »**. "
+                f"Son identifiant est le €{cagnotte_info.id}"
+            )
+
+            await update_forbes_classement(message.guild, self, self.client)
+
+        elif "créer $tyle" in message_command:
+            cagnotte_name = " ".join(splited_command[3:])
+            if len(cagnotte_name) == 0:
+                await message.channel.send(
+                    "Merci de mentionner un nom pour ta €agnotte."
+                )
+                return
+            self.swag_bank.create_cagnotte(
+                cagnotte_name, Currency.STYLE, message.author.id
+            )
+
+            cagnotte_info = self.swag_bank.swagdb.get_cagnotte(cagnotte_name).get_info()
+            await message.channel.send(
+                f"{message.author.mention} vient de créer une €agnotte de $tyle nommée **« {cagnotte_name} »**. "
+                f"Son identifiant est le €{cagnotte_info.id}"
+            )
+            await update_forbes_classement(message.guild, self, self.client)
+
+        elif "créer" in splited_command:
+            await message.channel.send(
+                "Merci de mentionner le type de monnaie de la €agnotte "
+                "après le mot clef **créer**"
+            )
+
+        elif set(splited_command).intersection(
```

Comment:
Ok là je suis du-per, what does this do?

Reply:
Ouais, alors là c'est un peu perché :

En gros, on vérifie si la commande demandé par l'utilisateur est une commande qui a besoin d'un index de €agnotte (rajouté par le "€" (€2 par exemple)).
Ces commandes sont les commandes {"info", "historique", "payer", "donner", "loto", "partager", "détruire"}

Si l'utilisateur n'a pas mentionné de €agnotte, mais qu'il souhaitait utiliser l'une de ces commandes, alors on lui envoie le message "il manque l'identifiant de la €agnotte etc.".

Si l'utilisateur ne mentionne aucune de ces commandes, alors il faut afficher la liste des commandes disponibles pour les €agnottes (cf le dernier if)

Reply:
Ok, donc c'est plutôt la fonction `get_cagnotte_idx_from_command` devrait raise une `CagnotteUnspecifiedException` ou un truc du genre qui devrait être attrapé par le try-except général du module qui lui devrait envoyer le message d'erreur.
Sinon 1. la gestion des message d'erreur va se retrouver un peu mélangée un peu partout et 2. cette ligne va être un peu compliquée à relire à l'avenir
Ah, et 3. `not any(argument.startswith("€") for argument in splited_command[1:])` me paraît plus clean, si jamais tu as besoin d'un test pour raise l'exception de manière safe ?

Reply:
not any(argument.startswith("€") for argument in splited_command[1:]) is not enough because if the user say  `!€agnotte bullshit` it won't write the help message with all the correct commands

Reply:
Sorry, I didn't get what you just said

But, I mean— if `get_cagnotte_id` raises an exception if it doesn't find a valid €id (aka a `s: str` with `s.startswith("€") is True` and `int(s[1:])` doesn't raise a `ValueError` or whatever it blrups on invalid `str` input), you don't need this akward test anymore, do you?

Reply:
OOOH DAAAAMNNHHH you're right.

Reply:
Resolved here : https://github.com/BobbySoftwares/Bybbot/pull/7/commits/e03e6ef1230315363c1a57ecc4a52ecba6341842